BACnetClient.cs

```diff
@@ -560,34 +554,48 @@ protected void ProcessSegmentAck(BacnetAddress address, BacnetPduTypes type, byt
             }
         }
 
-        public delegate void SegmentHandler(BacnetClient sender, BacnetAddress address, BacnetPduTypes type, BacnetConfirmedServices service, byte invokeId, BacnetMaxSegments maxSegments, BacnetMaxAdpu maxAdpu, byte sequenceNumber, bool first, bool moreFollows, byte[] buffer, int offset, int length);
+        public delegate void SegmentHandler(BacnetClient sender, BacnetAddress address, BacnetPduTypes type, BacnetConfirmedServices service, byte invokeId, BacnetMaxSegments maxSegments, BacnetMaxAdpu maxAdpu, byte sequenceNumber, byte[] buffer, int offset, int length);
         public event SegmentHandler OnSegment;
 
+        /// <summary>
+        /// Dictionary of List of Tuples with sequence-number and byte[] per invoke-id
+        /// TODO: invoke-id should be PER (remote) DEVICE!
+        /// </summary>
+        private Dictionary<byte, List<Tuple<byte, byte[]>>> segmentsPerInvokeId = new Dictionary<byte, List<Tuple<byte, byte[]>>>();
+        private Dictionary<byte, object> locksPerInvokeId = new Dictionary<byte, object>();
+        private Dictionary<byte, byte> expectedSegmentsPerInvokeId = new Dictionary<byte, byte>();

```

Comment:
yeah... about underscore prefix: I don't like it - I find that it "breaks the flow" when reading code - and it looks so 1970.
.NET Framework Design Guidelines state _"**DO NOT** use underscores, hyphens, or any other nonalphanumeric characters"_.
And yes, I am painfully aware that in .NET Core, there are underscores all over the place - that's what you get when you take it to "the community" I guess.

Why would you use an underscore to indicate private fields, but not use i_ThisIsAnInt or m_ThisIsAMember? Just tell ReSharper to give private members a different color then local variables and be done with it 😉

Reply:
I didn't like underscore too because in the past after brief rendezvous with Pascal and Delphi I was raised with C# and underscore was not present in the convention. But I also didn't like to write ```this.``` everywhere when the method argument is named the same. What is a deal breaker for now is that it's the convention here already so please adapt, later when a refactor is made on everything, this can be changed. More than personal preference (mine or yours) I value consistency.